manim/scene/scene.py

```diff
@@ -832,6 +848,69 @@ def update_skipping_status(self):
                 file_writer_config["skip_animations"] = True
                 raise EndSceneEarlyException()
 
+    def handle_caching_play(func):
+        """
+        This method is used internally to wrap the passed function into a function that will compute the hash of the play invocation, 
+        and will act accordingly : either skip the animation because already cached, either nothing and let the play invocation be processed normally.
+
```

Comment:
Suggestion: please stick to the "one line plus an explanatory (optional) paragraph" format. I suggest

```
Returns a wrapped version of func that will compute the hash of the play invocation.

The returned function will act according to the computed hash: either skip the animation because it's already cached, or let the invoked function play normally.
```

Reply:
Also, I just realized this and the next function are (and are used as) decorators. So their docstrings should mention that.